<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>897</width>
    <height>694</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>897</width>
    <height>694</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>897</width>
    <height>694</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="minimumSize">
    <size>
     <width>897</width>
     <height>694</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>897</width>
     <height>16777215</height>
    </size>
   </property>
   <widget class="QWidget" name="toolBar" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>901</width>
      <height>71</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: #696969;  
border: 2px solid #505050;  
border-radius: 8px;  </string>
    </property>
    <widget class="QPushButton" name="connectTcpButton">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>111</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
     <property name="text">
      <string>CONNECT</string>
     </property>
    </widget>
    <widget class="QPushButton" name="load_fButton">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>10</y>
       <width>88</width>
       <height>25</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; 
</string>
     </property>
     <property name="text">
      <string>LOAD FILE</string>
     </property>
    </widget>
    <widget class="QPushButton" name="save_fButton">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>40</y>
       <width>88</width>
       <height>25</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
     <property name="text">
      <string>SAVE FILE</string>
     </property>
    </widget>
    <widget class="QPushButton" name="exitButton">
     <property name="geometry">
      <rect>
       <x>760</x>
       <y>10</y>
       <width>121</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
     <property name="text">
      <string>EXIT</string>
     </property>
    </widget>
    <widget class="QDial" name="x_dial">
     <property name="geometry">
      <rect>
       <x>710</x>
       <y>30</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-100</number>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
    </widget>
    <widget class="QDial" name="y_dial">
     <property name="geometry">
      <rect>
       <x>670</x>
       <y>30</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-100</number>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
    </widget>
    <widget class="QDial" name="dial">
     <property name="geometry">
      <rect>
       <x>589</x>
       <y>23</y>
       <width>41</width>
       <height>51</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-100</number>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
    </widget>
    <widget class="QDial" name="dial_2">
     <property name="geometry">
      <rect>
       <x>629</x>
       <y>23</y>
       <width>41</width>
       <height>51</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-100</number>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>680</x>
       <y>10</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
     <property name="text">
      <string>IMU Ctrl</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>590</x>
       <y>10</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
     <property name="text">
      <string>PWM Ctrl</string>
     </property>
    </widget>
    <widget class="QPushButton" name="tof1_button">
     <property name="geometry">
      <rect>
       <x>500</x>
       <y>40</y>
       <width>88</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>tof1</string>
     </property>
    </widget>
    <widget class="QPushButton" name="line1_button">
     <property name="geometry">
      <rect>
       <x>500</x>
       <y>10</y>
       <width>88</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>lin1</string>
     </property>
    </widget>
    <widget class="QPushButton" name="disconnectTcpButton">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>10</y>
       <width>111</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
     <property name="text">
      <string>DISCONNECT</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="IpText">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>10</y>
       <width>113</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="PortText">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>40</y>
       <width>113</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="graphsView" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>70</y>
      <width>401</width>
      <height>621</height>
     </rect>
    </property>
    <property name="maximumSize">
     <size>
      <width>401</width>
      <height>621</height>
     </size>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LayoutDirection::LeftToRight</enum>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: #bbbbbb;  
border: 2px solid #505050;  
border-radius: 8px;  </string>
    </property>
    <widget class="QLabel" name="imu_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>381</width>
       <height>17</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
     <property name="text">
      <string>IMU</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
    <widget class="QWidget" name="imu_widget" native="true">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>30</y>
       <width>341</width>
       <height>131</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
    </widget>
    <widget class="QWidget" name="motors_widget" native="true">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>200</y>
       <width>341</width>
       <height>131</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
    </widget>
    <widget class="QLabel" name="motors_speed_la">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>180</y>
       <width>381</width>
       <height>17</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
     <property name="text">
      <string>MOTORS SPEED</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
    <widget class="QWidget" name="lineS_widget" native="true">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>380</y>
       <width>341</width>
       <height>81</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
    </widget>
    <widget class="QLabel" name="lineS_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>360</y>
       <width>381</width>
       <height>17</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
     <property name="text">
      <string>LINE SENSORS</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
    <widget class="QWidget" name="tof_widget" native="true">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>510</y>
       <width>341</width>
       <height>81</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
    </widget>
    <widget class="QLabel" name="tof_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>490</y>
       <width>381</width>
       <height>17</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
     <property name="text">
      <string>REFLECTIVE SENSORS</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="robotView" native="true">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>70</y>
      <width>501</width>
      <height>621</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: #555555;
border: 2px solid #505050;  
border-radius: 8px;  
</string>
    </property>
    <widget class="QLabel" name="motor_speed_1_Label">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>490</y>
       <width>61</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
     <property name="text">
      <string>10%</string>
     </property>
     <property name="scaledContents">
      <bool>false</bool>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="motor_speed_2_Label">
     <property name="geometry">
      <rect>
       <x>330</x>
       <y>490</y>
       <width>61</width>
       <height>31</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: #B0B0B0;
color: #555555;            
font-weight: bold;
border: 2px solid #505050; </string>
     </property>
     <property name="text">
      <string>10%</string>
     </property>
     <property name="scaledContents">
      <bool>false</bool>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="mainrobot">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>340</y>
       <width>131</width>
       <height>131</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/robotv1.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
     <property name="openExternalLinks">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="lineS_Active_1">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>340</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:0px;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/activeSensor.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="lineS_Active_2">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>340</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:0px;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/activeSensor.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="lineS_Active_3">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>440</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:0px;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/activeSensor.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="tof_Green_1">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>340</y>
       <width>151</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:0px;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/greenLEFT.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="tof_Green_2">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>160</y>
       <width>71</width>
       <height>171</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:0px;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/greenUP.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="tof_Green_3">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>160</y>
       <width>71</width>
       <height>171</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:0px;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/greenUpR.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="tof_Green_4">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>340</y>
       <width>141</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:0px;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/greenRIGHT.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="tof_Red_1">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>340</y>
       <width>151</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:0px;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/redLEFT.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="tof_Red_2">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>160</y>
       <width>71</width>
       <height>171</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:0px;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/redUp.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="tof_Red_3">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>160</y>
       <width>71</width>
       <height>171</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:0px;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/redUpR.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="tof_Red_4">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>340</y>
       <width>141</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:0px;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/redRIGHT.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="motor1_arrow">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>390</y>
       <width>31</width>
       <height>91</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:0px;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/arrowUp.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="motor2_arrow">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>390</y>
       <width>31</width>
       <height>91</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border:0px;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/arrowUp.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="imu_arrow">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>320</y>
       <width>181</width>
       <height>181</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background: transparent;
border: none;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/arrowIMU.png</pixmap>
     </property>
    </widget>
    <widget class="QLabel" name="connectGreen">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>20</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background: transparent;
border: none;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/connectionGreen.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="connectRed">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>20</y>
       <width>51</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background: transparent;
border: none;</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="resources.qrc">:/new/image/img/connectionRed.png</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>exitButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>737</x>
     <y>29</y>
    </hint>
    <hint type="destinationlabel">
     <x>798</x>
     <y>538</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
